src/EditorFeatures/CSharpTest2/Recommendations/ThisKeywordRecommenderTests.cs

```diff
@@ -632,5 +632,71 @@ void Main(string[] args)
 }
 ", matchPriority: SymbolMatchPriority.Keyword);
         }
+
+        [Fact]
+        public async Task TestExtensionMethods_FirstParameter_AfterRefKeyword()
+        {
+            await VerifyKeywordAsync(@"
+public static class Extensions
+{
+    public static void Extension(ref $$ object obj, int x) { }
+}
+");
+        }
+
+        [Fact]
+        public async Task TestExtensionMethods_FirstParameter_AfterRefReadOnlyKeywords()
+        {
+            await VerifyKeywordAsync(@"
+public static class Extensions
+{
+    public static void Extension(ref readonly $$ object obj, int x) { }
+}
+");
+        }
+
+        [Fact]
+        public async Task TestExtensionMethods_FirstParameter_AfterInKeyword()
+        {
+            await VerifyKeywordAsync(@"
+public static class Extensions
+{
+    public static void Extension(in $$ object obj, int x) { }
```

Comment:
No. But I'll add tests for that case, also for a case where it is in a non static class or a non static function.

---
In reply to: [120768100](https://github.com/dotnet/roslyn/pull/19703#discussion_r120768100) [](ancestors = 120768100)